witness/tests/Test/Dscp/Witness/Explorer/ExplorerSpec.hs

```diff
@@ -51,12 +53,16 @@ createAndSubmitPub genSecret = do
 
 -- | Dump all mempool transactions into a new block.
 dumpBlock
-    :: (WitnessWorkMode ctx m, WithinWriteSDLock)
-    => SlotId -> m HeaderHash
-dumpBlock slotId = do
-    block <- createBlock runSdM slotId
-    void $ applyBlock block
-    return (headerHash block)
+    :: (TestWitnessWorkMode ctx m, WithinWriteSDLock)
+    => m HeaderHash
+dumpBlock = do
+    slotId <- rewindToNextSlot
+    let issuerKey = KeyResources . mkSecretKeyData $ testFindSlotOwner slotId
```

Comment:
This operation is a bit uncontrollable. I don't think anything can happen so we might want to cancel this, though.

Reply:
Well, that's true. I'd like to come up with a better thing here, but with interface provided by our `witness <-> slot` mapping logic it's probably the only normal way.

Reply:
No, I'm about `createBlock` after `sleep`.

Reply:
Lool, that wasn't obvious for me.
Okay, I'll just probably tell in the comment that it's part of `rewindToNextSlot`'s semantics to return _new_ slot. If we cannot guarantee this simple invariant for `rewindToNextSlot`, then the whole mocking doesn't look sound.